openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/OpenEdgeComponents.java

```diff
@@ -229,32 +230,32 @@ private static void configureField(Object check, Field field, String value) {
 
       if (field.getType().equals(String.class)) {
         field.set(check, value);
-      } else if ("int".equals(field.getType().getSimpleName())) {
+      } else if (int.class == field.getType()) {
         field.setInt(check, Integer.parseInt(value));
-      } else if ("short".equals(field.getType().getSimpleName())) {
+      } else if (short.class == field.getType()) {
         field.setShort(check, Short.parseShort(value));
-      } else if ("long".equals(field.getType().getSimpleName())) {
+      } else if (long.class == field.getType()) {
         field.setLong(check, Long.parseLong(value));
-      } else if ("double".equals(field.getType().getSimpleName())) {
+      } else if (double.class == field.getType()) {
         field.setDouble(check, Double.parseDouble(value));
-      } else if ("boolean".equals(field.getType().getSimpleName())) {
+      } else if (boolean.class == field.getType()) {
         field.setBoolean(check, Boolean.parseBoolean(value));
-      } else if ("byte".equals(field.getType().getSimpleName())) {
+      } else if (byte.class == field.getType()) {
         field.setByte(check, Byte.parseByte(value));
-      } else if (field.getType().equals(Integer.class)) {
+      } else if (Integer.class == field.getType()) {
         field.set(check, new Integer(Integer.parseInt(value)));
-      } else if (field.getType().equals(Long.class)) {
+      } else if (Long.class == field.getType()) {
         field.set(check, new Long(Long.parseLong(value)));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Long" constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2129)